### PR TITLE
Add quarkus:dev option for running the examples

### PR DESCRIPTION
**Pull request by tplevko**
Based on [comment](https://github.com/apache/camel-k-runtime/pull/643#issuecomment-804727535) - adding the option to run the examples using:

`mvn clean compile quarkus:dev`

@lburgazzoli there seems to be a problem with the java example, which doesn't seem to run using the above mentioned command - there seems to be a problem with classloading, the problem is:

```
2021-04-16 13:51:22,023 ERROR [org.apa.cam.qua.mai.CamelMainRuntime] (Quarkus Main Thread) Failed to start application: org.apache.camel.RuntimeCamelException: org.joor.ReflectException: Compilation error: /MyRoutes.java:17: error: package org.apache.camel.builder does not exist
import org.apache.camel.builder.RouteBuilder;
                               ^
/MyRoutes.java:19: error: cannot find symbol
public class MyRoutes extends RouteBuilder {
                              ^
  symbol: class RouteBuilder
2 errors
```

I couldn't find the cause of this issue, I can however remove the option from this example, in case there is no easy solution.

**Release Note**
```release-note
NONE
```

### CODE REVIEW

File: examples/cron/pom.xml [hunk 1]
```diff
@@ -31,6 +31,7 @@
     <properties>
         <noDeps>true</noDeps>
         <quarkus.camel.main.routes-discovery.enabled>false</quarkus.camel.main.routes-discovery.enabled>
+        <quarkus.package.type>uber-jar</quarkus.package.type>
```
Comment (lburgazzoli) [inline examples/cron/pom.xml:34]: Is this change required ?

Comment (tplevko) [reply examples/cron/pom.xml:34]: Not necessarily. It is related to the PR I made previously and it effectively reverts the change I introduced, generating `uber-jar` instead of the `fast-jar`:

    java -jar target/quarkus-app/quarkus-run.jar
    java -jar target/camel-k-runtime-example-cron-runner.jar

In case this is undesired, I will remove it from the PR (Apparently I should have separated it in 2 commits in the first place.)

Comment (lburgazzoli) [reply examples/cron/pom.xml:34]: I'd prefer to use the fast-jar mode as it is the mode we use on camel-k (unless there are known issues)

Comment (tplevko) [reply examples/cron/pom.xml:34]: Understood, will remove this particular type of changes

Comment (tplevko) [reply examples/cron/pom.xml]: Should be back to the previous fast-jar usage

**Review by lburgazzoli — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]
